libreoffice: updates + updateScript + reformat

doronbehar opened the pull request:
## Description of changes

## Things done

**Haven't tested that it builds yet! Only tested that `configurePhase` is working.**

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

doronbehar committed:
libreoffice: improve nixpkgs' README update instructions

doronbehar committed:
libreoffice: small reformatting to postConfigure

doronbehar committed:
libreoffice: get rid of src-$VARIANT/override.nix

Towards simplfying the expression.

doronbehar committed:
libreoffice: always use stdenv.mkDerivation

`mkDerivation` from `libsForQt5` is deprecated. Hooks are preferred, see
https://github.com/NixOS/nixpkgs/issues/180841 .

doronbehar committed:
libreoffice: use finalAttrs in mkDerivation

doronbehar committed:
libreoffice: write a whole, batteries included updateScript

doronbehar committed:
libreoffice: reorder/reformat buildInputs a bit

The versionAtleast 7.4 is no longer needed due to still and fresh
variants' versions.

doronbehar committed:
mdds: 2.0.3 -> 2.1.1

Diff: https://gitlab.com/mdds/mdds/-/compare/2.0.3...2.1.1

Changelog: https://gitlab.com/mdds/mdds/-/blob/2.1.1/CHANGELOG

doronbehar committed:
libreoffice: reformat a bit kdeIntegration postPatch hook

Put the xdg-open -> kde-open5 substitution in the kdeIntegration
conditioned block, also put the comment about it from above the
postPatch near there.

7c6f434c commented:
Hmm fails on Linux…

doronbehar commented:
> Hmm fails on Linux…

Could you share the build log please? I don't have a strong machine to test the build with yet.

7c6f434c commented:
I meant on ofBorg. https://logs.ofborg.org/?key=nixos/nixpkgs.257853&attempt_id=32ac9f11-6a61-443b-b237-d01faf866acd

```
types.cpp: In function 'std::vector<std::pair<std::basic_string_view<char>, orcus::dump_format_t> > orcus::get_dump_format_entries()':
types.cpp:320:52: error: 'const struct mdds::chars_map_entry<orcus::dump_format_t, long unsigned int>' has no member named 'keylen'
  320 |         ret.emplace_back(std::string_view{e.key, e.keylen}, e.value);
      |                                                    ^~~~~~
types.cpp:320:58: error: no matching function for call to 'std::basic_string_view<char>::basic_string_view(<brace-enclosed initializer list>)'
  320 |         ret.emplace_back(std::string_view{e.key, e.keylen}, e.value);
      |                                                          ^
In file included from /nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/bits/basic_string.h:47,
                 from /nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string:53,
                 from ../../include/orcus/types.hpp:22,
                 from types.cpp:8:
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:139:7: note: candidate: 'constexpr std::basic_string_view<_CharT, _Traits>::basic_string_view(const _CharT*, size_type) [with _CharT = char; _Traits = std::char_traits<char>; size_type = long unsigned int]'
  139 |       basic_string_view(const _CharT* __str, size_type __len) noexcept
      |       ^~~~~~~~~~~~~~~~~
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:139:7: note:   conversion of argument 2 would be ill-formed:
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:133:7: note: candidate: 'constexpr std::basic_string_view<_CharT, _Traits>::basic_string_view(const _CharT*) [with _CharT = char; _Traits = std::char_traits<char>]'
  133 |       basic_string_view(const _CharT* __str) noexcept
      |       ^~~~~~~~~~~~~~~~~
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:133:7: note:   candidate expects 1 argument, 2 provided
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:130:17: note: candidate: 'constexpr std::basic_string_view<_CharT, _Traits>::basic_string_view(const std::basic_string_view<_CharT, _Traits>&) [with _CharT = char; _Traits = std::char_traits<char>]'
  130 |       constexpr basic_string_view(const basic_string_view&) noexcept = default;
      |                 ^~~~~~~~~~~~~~~~~
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:130:17: note:   candidate expects 1 argument, 2 provided
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:126:7: note: candidate: 'constexpr std::basic_string_view<_CharT, _Traits>::basic_string_view() [with _CharT = char; _Traits = std::char_traits<char>]'
  126 |       basic_string_view() noexcept
      |       ^~~~~~~~~~~~~~~~~
/nix/store/h5kvfrjmpw792v8jg7nrzfkffmn0iyy8-gcc-12.3.0/include/c++/12.3.0/string_view:126:7: note:   candidate expects 0 arguments, 2 provided
```

doronbehar commented:
> I meant on ofBorg.

Thanks, It was related to the mdds 2.0 -> 2.1 substitution in configure.ac... I switched to using the system `mdds` only for the "fresh" variant.

7c6f434c commented:
@ofborg build libreoffice-still libreoffice-fresh

doronbehar committed:
libreoffice: remove stale, unused patches

doronbehar committed:
libreoffice-fresh: disable a failing test

doronbehar committed:
libreoffice: disable a test failing on aarch64

doronbehar commented:
@ofborg build libreoffice-still
@ofborg build libreoffice-fresh
@ofborg build libreoffice-qt

Hopefully when ofborg gets separate commands, it will create a separate job for each variant.

doronbehar commented:
CI is green. I'll give this a week or so to lie down. Reviews are welcome - changes are robust overall.

7c6f434c commented:
I assume that the test-blocking changes are just a reformatting…

Changes look good to me.

nadir-yttrium commented:
Seems I'm too late, only just finished my build attempt  of libreoffice-fresh when the latest commit was a9b75ec64ed1fbcec309929f98c586bac5399b1d. 

It failed for me. Logs: 

https://gist.github.com/nadir-yttrium/5c13025420dcf00effa56496a2f32138

Took my machine ages. I can probably give it another try tomorrow with the up to date commit.

doronbehar commented:
> Logs:

That log seems incomplete... Never mind, it likely failed as it failed on ofborg before. I forgot about the option to use ofborg to test it compiles. Now you should be able to successfully build it and test it's functionality. I have a strong machine in my office I will be able to use only on Monday that can build this, but having someone else test functionality would be nice.

AndersonTorres commented on pkgs/applications/office/libreoffice/update.sh:
```diff
@@ -0,0 +1,72 @@
+# then extract the `download.lst` file for ../generate-libreoffice-srcs.py.
+main_path_hash=($(nix-prefetch-url --print-path $baseurl/libreoffice-$full_version.tar.xz))
+echo "{" > main.nix
+echo "  sha256 = "\"${main_path_hash[0]}'";' >> main.nix
```
it can be `hash` instead.

doronbehar replied:
> it can be `hash` instead.

I was thinking about it, but apparently `nix-prefetch-url` prints the sha256 in base 16, and not 64. I'm not sure how to convert between the two, and I wanted to have consistency among all files `main.nix` `help.nix` and `translations.nix`. `nix-prefetch` writes `sha256` and not `hash`, but uses base 64 and I couldn't find a way to change that behavior.

AndersonTorres replied:
Why not to get rid of `nix-prefetch`, then? The main repo looks abandoned, and the precise pull request that would allow `hash` was not accepted more than one year.

https://github.com/msteen/nix-prefetch/pull/43

doronbehar replied:
> Why not to get rid of `nix-prefetch`, then? The main repo looks abandoned, and the precise pull request that would allow `hash` was not accepted more than one year.
> 
> [msteen/nix-prefetch#43](https://github.com/msteen/nix-prefetch/pull/43)

You have a point. Still though, `nix-prefetch-url` always uses `sha256` in base 16, so I'll have to use `sha256` key in the generated nix files.

I fixed this in a rebase.

doronbehar committed:
libreoffice-still-unwrapped: 7.4.7.1 -> 7.5.7.1

Remove dolphin templates, as these seem empty now (for libreoffice-fresh
as well).

doronbehar committed:
libreoffice-fresh-unwrapped: 7.5.4.2 -> 7.6.2.1

doronbehar commented:
I built the packages, and noticed an outdated section in the `postInstall`, not critical for functionality. Other then that I tested openning a document etc and it seems fine. Will merge when CI is green.

doronbehar commented:
Pushed another small fixup - removing the dolphin templates - they seem to not exist anymore on upstream's source.
